Sources/StreamChat/Utils/KeyedDecodingContainer+Array.swift

```diff
@@ -6,25 +6,83 @@ import Foundation
 
 private struct ElementWrapper<T: Decodable>: Decodable {
     let value: T?
+    var rawJSON: [String: RawJSON]?
     var error: Error?
     init(from decoder: Decoder) throws {
         do {
             value = try T(from: decoder)
         } catch {
             value = nil
             self.error = error
+            // We decode the payload as RawJSON to be able to display the payload in case of error
+            rawJSON = try? [String: RawJSON](from: decoder)
```

Comment:
Can we measure if this adds a lof of overhead?

Reply:
Here's a test I've written to measure the impact of this PR, I'll put the test at the end of this comment so the measurements stay on top.

I've used a message array (since MessagePayload is quite big - impact should be big too), with 1000 messages.

(1) Using `decode` | (2) Using `decodeArrayIgnoringFailures` | (3) Using `decodeArrayIgnoringFailures` when `updated_at` is deleted from payload
---- | ----- | -----
460ms | 461ms | 533ms

Case (3) is the extreme case where every object fails to decode (hence they're decoded as RawJSON, encoded to String and printed to console) and the array is quite big, we never have this big arrays. Performance scales linearly so for an array of 100 MessagePayloads I've observed 46ms for (1). I can say that performance impact is around 10%, which is good enough for this kind of verbosity for our errors.

The test code:
```swift
func test_decode_withErroringPayloads() throws {
    struct MessageArrayContainer: Decodable {
        enum CodingKeys: String, CodingKey {
            case messages
        }
        
        let messages: [MessagePayload]
        
        init(from decoder: Decoder) throws {
            let container = try decoder.container(keyedBy: CodingKeys.self)
            
            messages = try container.decodeArrayIgnoringFailures([MessagePayload].self, forKey: .messages)
        }
    }
    
    let messagePayload = """
        {
        "id" : "broken-waterfall-5-7aede36b-b89f-4f45-baff-c40c7c1875d9",
        "reaction_counts" : {
        
        },
        "silent" : false,
        "created_at" : "2020-06-09T08:10:40.800912Z",
        "reaction_scores" : {
        
        },
        "type" : "regular",
        "latest_reactions" : [
        
        ],
        "text" : "sadfadf",
        "attachments" : [
        
        ],
        "own_reactions" : [
        
        ],
        "updated_at" : "2020-06-09T08:10:40.800912Z",
        "reply_count" : 0,
        "user" : {
        "id" : "broken-waterfall-5",
        "banned" : false,
        "unread_channels" : 0,
        "extraData" : {
        "name" : "Tester"
        },
        "totalUnreadCount" : 0,
        "last_active" : "2020-06-10T13:24:00.501797Z",
        "created_at" : "2019-12-12T15:33:46.488935Z",
        "unreadChannels" : 0,
        "unread_count" : 0,
        "image" : "https://getstream.io/random_svg/?id=broken-waterfall-5&amp;name=Broken+waterfall",
        "updated_at" : "2020-06-10T14:11:29.946106Z",
        "role" : "user",
        "total_unread_count" : 0,
        "online" : true,
        "name" : "broken-waterfall-5"
        },
        "mentioned_users" : [
        
        ],
        "html" : "<p>sadfadf</p>",
        "pinned": false,
        "pinned_at": null,
        "pin_expires": null,
        "pinned_by": null
        }
        """
    let messageArrayPayload = "{ \"messages\": [\(Array(repeating: messagePayload, count: 1000).joined(separator: ",\n"))]}"
    let messageArrayPayloadData = messageArrayPayload.data(using: .utf8)!
    
    measure {
        do {
            _ = try JSONDecoder.default.decode(MessageArrayContainer.self, from: messageArrayPayloadData)
        } catch {
            XCTFail("Failed to parse JSON: \(error)")
        }
    }
}
```

Reply:
Good to go then 🚀